src/utils/ProfileUtils.ts

```diff
@@ -125,6 +125,27 @@ export async function installModsToProfile(
 
 export async function parseYamlToExportFormat(yamlContent: string) {
     const parsedYaml = await yaml.parse(yamlContent);
+    if (!parsedYaml) {
+        throw new R2Error(
+            'Failed to parse yaml contents.',
+            'Yaml parsing failed when trying to import profile via file (The contents of export.r2x file are invalid).',
+            'Ensure that the profile import file isn\'t corrupted.'
+        )
+    }
+    if (!parsedYaml.profileName || typeof parsedYaml.profileName !== 'string') {
+        throw new R2Error(
+            'Failed to read profile name.',
+            'Reading the profile name after parsing the yaml failed (export.r2x is missing the profileName field).',
+            'Ensure that the profile import file isn\'t corrupted.'
+        )
+    }
+    if (!parsedYaml.mods || typeof parsedYaml.mods !== 'object') {
```

Comment:
Looking at old L130/new L151 it looks like we're expecting an array, and while object might be an array, it might not. Using `if(!Array.isArray(parsedYaml.mods) {` would be more accurate, shorter, and make the intent clearer to the reader.